Cabal/Distribution/PackageDescription/Check.hs

```diff
@@ -448,20 +448,20 @@ checkFields pkg =
            "Package names with the prefix 'z-' are reserved by Cabal and "
         ++ "cannot be used."
 
-  , check (isNothing (buildType pkg)) $
+  , check (isNothing (buildTypeRaw pkg) && specVersion pkg < mkVersion [2,1]) $
```

Comment:
This is great change, which one should mention in the changelog, It's updated in the manual (great!), but let's advertise this change in Changelog too.

Optionally, we can add a section in changelog gathering all cabal format related changes.

Reply:
Right!

Fwiw, I've got more related changes to for `cabal check` & build-type in the pipeline; the idea about structuring the changelog is a good one too, although I've been wanting to have an explicit cabal spec changelog in the user's guide as well... :-)